cardinal/world.go

```diff
@@ -207,6 +208,41 @@ func NewMockWorld(opts ...WorldOption) (*World, error) {
 	return world, nil
 }
 
+func RegisterMessage[In any, Out any](world *World, name string, opts ...message.MessageOption[In, Out]) error {
+	msgType := message.NewMessageType[In, Out](name, opts...)
+	var msg types.Message = msgType
+	err := world.registerMessagesByName(msg)
+	if err != nil {
+		return eris.Wrap(err, "failed to register message")
+	}
+	var ok bool
+	msg, ok = world.GetMessageManager().GetMessageByName(name)
+	if !ok {
+		return eris.Errorf("Failed to register message with name: %s", name)
+	}
+	typeValueOfMessageType := reflect.TypeOf(*msgType)
+	return world.GetMessageManager().RegisterMessageByType(typeValueOfMessageType, msg)
```

Comment:
i'm confused on the purpose of registering by type this way. we are storing by type of msgType, which is `MessageType`, so that people can get `types.Message` by type right? but if they have the `MessageType` already, then they have a `types.Message`?

Reply:
type.Message can't be keyed by type because it's just an interface. It's the same type. 

However MessageType[K, V] is different for all possible Ks and Vs. That's why it can be used as a key value here. 

The purpose of registering the type this way is such that it can be retrieved via a generic function:

`GetMessage[K, V](wCtx)` this is effectively what eliminates the global.

Reply:
Depending on the code, the other `map[string]types.Message` on the message manager will eventually be removed, once all dependencies and tests that utilize it are eliminated (if they can be at all).

Reply:
got it, missed the GetMessage[K, V](wCtx) part